pennylane/utils.py

```diff
@@ -433,3 +434,77 @@ def expand_vector(vector, original_wires, expanded_wires):
     expanded_tensor = np.moveaxis(expanded_tensor, original_indices, wire_indices)
 
     return expanded_tensor.reshape(2 ** M)
+
+
+class Spy:
+    """Context in which calls to a QubitDevice's execute method are counted without performing the execution.
+
+    The class mocks the execute method, returning a dummy value of the correct dimensions.
```

Comment:
I'm just trying to think if we can make this more explanatory for users seeing this as part of a summary table (so only the first line will show up).

```suggestion
class Spy:
    """Context in which calls to quantum devices are simply counted without performing the execution.

    The class mocks the execute method, returning a dummy value of the correct dimensions.
    
    .. note:: Currently, only qubit devices are supported.
```

Reply:
"Currently, only qubit devices are supported." is not quite as accurate as saying `QubitDevice`, but I figure to a lot of users, `QubitDevice` will not mean much. As far as I am aware, the only qubit device that doesn't inherit from `QubitDevice` is `default.tensor`.